### ISSUE TITLE
/en-US/docs/Web/HTTP/Headers/Cache-Control description of Cacheability is confusing

### ISSUE THREAD

**Issue by NeilMadden** [mdn/sprints#3257]
# Request type

- [ ] Please close this issue, I accidentally submitted it without adding any details
- [ ] New documentation
- [x] Correction or update

# Details

The MDN [page on Cache-Control](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#Cacheability) has a section on "Cacheability" that is very hard to parse at the moment. The current text reads:

> A response is normally cached by the browser if:
> * it has a status code of 301, 302, 307, 308, or 410 **and**
> * Cache-Control does not have no-store, or if proxy, does not have private **and**
> * Authorization is unset
> * either
>     * has a status code of 301, 302, 307, 308, or 410 or
>     * has public, max-age or s-maxage in Cache-Control or
>     * has Expires set

Is the nested list of conditions supposed to be an alternative to the first three? Is the first bullet point supposed to read "it **does not** have a status code of 301, 302, ..."?

Sorry that I can't make a more helpful suggestion for what the text should be, but I'm unable to work out what the correct description should be.

**Comment by chrisdavidmills** [comment]
@NeilMadden is right that this is confusing. @Elchi3, you wrote this page originally; any ideas here (I appeciate that this stuff was added later)? I looked through all the RFCs that define this behavior, and couldn't find anything similar to these decriptions.

**Comment by Elchi3** [comment]
Looks like a contributor updated this about a month ago to this description. See https://wiki.developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control$compare?locale=en-US&to=1616123&from=1616040

Will need to take a closer look how to unpack this.

**Comment by Jxck** [comment]
It better to update based on RFC7234 and RFC7231.

and This section should focus on **directives** but current section includes (incorrect) cacheable algorithm.
correct algorithm is in https://tools.ietf.org/html/rfc7234#section-3 but I don't think its necessary to place in this page because Cacheable is explicitly controlled by Cache-Control header.

If cacheable section is required in MDN it should be in https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching.
(and I also find that there are no implicit caching say Heuristic Cacheing section in this page)

So my proposal is below.

- remove incorrect algorithm from Cache-Control directive section
- add cacheable algorithm in Caching page
- add heuristic cache in Caching page

any comments ? or I think I can update based on above.

**Comment by hamishwillee** [comment]
@Jxck Thanks for your analysis. 
1. Agree. I have done this change. This should indeed be covered, if anywhere, in https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching

2. I am not *certain* we should "add cacheable algorithm" in [Caching](https://wiki.developer.mozilla.org/en-US/docs/Web/HTTP/Caching) page. If you look at that doc (almost) all this this information is captured - just not summarized. 
  
   If we want to do this we perhaps turn [targets of caching operations](https://wiki.developer.mozilla.org/en-US/docs/Web/HTTP/Caching#Targets_of_caching_operations) into this overview of what gets cached and/or what does not get cached. In any case lets consider renaming this because it doesn't feel like the title describes the content. 

3. The heuristics stuff is covered in the "Freshness section" - search for "Finally, if neither header is present, look". I think it would help to use the text heuristics in there so that readers at least can see the mapping to that section in the spec. How about putting that paragraph in a sub-section?

4. We should add s-max-age into the docs as that is not mentioned. At some point a re-validation against the spec to see we didn't miss anything else. 

5. This whole page needs a little subedit for style; the English is accurate, but parts are not particularly readable for non-native speakers.

**Comment by Jxck** [comment]
@hamishwillee 

1. LGTM

2. I suggested adding algorithm because of it used to has incorrect one. but I also think it reasonable to remove it from MDN.

3. LGTM

4. I think so

5. I think so

Totally, I'm agree with revise whole article related caching and revalidate with spec. Because en-US version is a source of other translated version, it more important.

I heard that PR reviewing process will be running after migrating.
so is it better to send a PR for each step and running reviewing for here after review process start ?

**Comment by hamishwillee** [comment]
> I heard that PR reviewing process will be running after migrating.
> so is it better to send a PR for each step and running reviewing for here after review process start ?

I think it is worth creating a separate issue for s-max-age and for the overall review and subedit. If we are able to look at this page fore we have git based source these can be handled just as this issue has been. If we have git-based source then we can use PRs to fix. Either way, we start with the issue. 

If you want to post those let me know. Otherwise I'll do that and close this on Friday.

**Comment by Jxck** [comment]
Thanks. Please do that, I'll follow you.

**Comment by hamishwillee** [comment]
Thanks @Jxck.  
- I updated doc to split this the information about heuristic approaches to freshness checking into its own section as discussed: https://wiki.developer.mozilla.org/en-US/docs/Web/HTTP/Caching#Heuristic_freshness_checking (item 3 above). FYI @chrisdavidmills 

- Issue https://github.com/mdn/sprints/issues/3864 created to track future spec update/subedit.

Closing this now as discussed
